components/arm/arm.go

```diff
@@ -415,3 +435,32 @@ func GoToWaypoints(ctx context.Context, a Arm, waypoints [][]referenceframe.Inpu
 	}
 	return nil
 }
+
+// checkDesiredJointPositions validates that the desired joint positions either bring the joint back
+// in bounds or do not move the joint more out of bounds.
+func checkDesiredJointPositions(ctx context.Context, a Arm, desiredJoints []float64) error {
```

Comment:
Each individual arm driver should call this on MoveToJointPositions (and it should be removed from `reconfigurableArm`